dispatcher/nm-dispatcher.c

```diff
@@ -627,6 +619,26 @@ find_scripts (const char *str_action)
 	}
 	g_dir_close (dir);
 
+out:
+	return sorted;
+}
+
+static GSList *
+find_scripts (const char *str_action)
+{
+	GSList *sorted = NULL;
+	const char *subdir = NULL;
+
+	if (   strcmp (str_action, NMD_ACTION_PRE_UP) == 0
+	    || strcmp (str_action, NMD_ACTION_VPN_PRE_UP) == 0)
+		subdir = "pre-up.d";
+	else if (   strcmp (str_action, NMD_ACTION_PRE_DOWN) == 0
+	         || strcmp (str_action, NMD_ACTION_VPN_PRE_DOWN) == 0)
+		subdir = "pre-down.d";
+
+	sorted = _find_scripts (sorted, NMCONFDIR, subdir);
+	sorted = _find_scripts (sorted, NMLIBDIR, subdir);
```

Comment:
while at it, why not also check `/run/NetworkManager/dispatcher.d`? It seems to add no extra effort.

Reply:
I'm struggling to figure out an use for it that wouldn't be sort of stupid. I'd prefer if someone actually needed it first.

Reply:
OK, makes sense.